admin_guide/image_policy.adoc

```diff
@@ -0,0 +1,154 @@
+[[admin-guide-image-policy]]
+= Image Policy
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+[[image-policy-overview]]
+== Overview
+
+You can control which images are allowed to run on your cluster using the ImagePolicy
+admission plugin (currently considered beta).  It allows you to control:
+
+ 1. the source of images: which registries can be used to pull images
+ 2. image resolution: force pods to run with immutable digests to ensure the image doesn't change due to a re-tag
+ 3. docker image label restrictions: force an image to have or not have particular labels
+ 4. image annotation restrictions: force an image in the integration docker registry to have or not have particular annotations
+
+
+== Configuring the ImagePolicy admission plugin
+
+To enable this feature, you'll need to configure the plugin in `master-config.yaml`.
+Here is an annotated example file.
+
+====
+
+[source,yaml]
+----
+
+admissionConfig:
+  pluginConfig:
+    openshift.io/ImagePolicy:
+      configuration:
+        kind: ImagePolicyConfig
+        apiVersion: v1
+        resolveImages: AttemptRewrite <1>
+        executionRules: <2>
+        - name: execution-denied
+          # Reject all images that have the annotation images.openshift.io/deny-execution set to true.
+          # This annotation may be set by infrastructure that wishes to flag particular images as dangerous
+          onResources: <3>
+          - resource: pods
+          - resource: builds
+          reject: true <4>
+          matchImageAnnotations: <5>
+          - key: images.openshift.io/deny-execution
+            value: "true"
+          skipOnResolutionFailure: true <6>
+        - name: allow-images-from-internal-registry
+          # allows images from the internal registry and tries to resolve them
+          onResources:
+          - resource: pods
+          - resource: builds
+          matchIntegratedRegistry: true
+        - name: allow-images-from-dockerhub
+          onResources:
+          - resource: pods
+          - resource: builds
+          matchRegistries:
+          - docker.io
+
+----
+
+<1> Try to resolve images to an immutable image digest and update the in the pod
+<2> Array of rules to evaluate against incoming resources.  If you only have reject==true rules
+    the default is "allow all".  If you have any accept rule, the default is "deny all".
+<3> Indicates which resources to enforce rules upon.  If nothing is specified, the default is "pods"
+<4> Indicates that if this rule matches, the pod should be rejected
+<5> List of annotations to match on the Image object's metadata
+<6> If you aren't able to resolve the image, don't fail the pod
+
+====
+
+
```

Comment:
We need to add an anchor just above the heading, so:
[[image-policy-testing-image-policy-admission-plug-in]]